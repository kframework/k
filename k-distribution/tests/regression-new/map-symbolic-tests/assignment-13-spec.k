// Copyright (c) 2019 K Team. All Rights Reserved.

requires "map-tests.k"

module ASSIGNMENT-13-SPEC
    imports MAP-TESTS

    rule <k> assignment ( (MAP:Map X:MyId |-> 1) [ Y:MyId <- 2 ] [ Z:MyId <- 3 ] ) => assignmentResult (   MAP [ Y <- 2 ]  X |-> 3              )   </k> requires Y =/=K X andBool Z =/=K Y andBool Z  ==K X

endmodule
