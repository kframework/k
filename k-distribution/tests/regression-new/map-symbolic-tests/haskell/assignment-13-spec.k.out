  #Not ( {
    Y
  #Equals
    Z
  } )
#And
  <k>
    assignmentResult ( MAP [ Y:MyId <- 2 ]
    Z:MyId |-> 3 ) ~> .
  </k>
#And
  {
    Z:MyId in_keys ( MAP )
  #Equals
    false
  }
